server/api/branch/tests.py

```diff
@@ -33,3 +37,53 @@ def test_GetAllExceptis_deleted(self):
         self.branch.delete()
         self.branch2.delete()
         self.assertEqual(len(Branch.objects_deleted.all()), 2)
+
+
+ADMIN = "adminTest2"
+ADMIN_PASS = "adminTestPassword2"
+
+
+USER = "normalUser2"
+USER_PASS = "normalUserPassword2"
+
+
+class BranchAuthTest(TestCase):
+
+    def setUp(self) -> None:
+        self.client = APIClient()
+        self.adminUser: User = User.objects.create_user(
+            username=ADMIN, password=ADMIN_PASS, is_staff=True
+        )
+        self.url = reverse("branch-list")
+        self.user: User = User.objects.create_user(username=USER, password=USER_PASS)
+        self.postData = {
+            "name": "My Branch",
+            "description": "This is a description of my Branch.",
+        }
+
+    def test_unauthorized(self):
+        self.client.login(username=USER, password=USER_PASS)
+        response = self.client.post(self.url, data=self.postData, format="json")
+        getResponse = self.client.get(self.url)
+        self.assertEqual(getResponse.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.client.logout()
+
+    def test_authorized(self):
+        self.client.login(username=ADMIN, password=ADMIN_PASS)
+        response = self.client.post(self.url, data=self.postData, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.client.logout()
```

Comment:
The naming of these is confusing, because you're actually testing permissions instead of auth. A `test_unauthorized` should test for a 401 if the user is not logged in at all